src/main/java/org/sagebionetworks/web/client/widget/sharing/SharingPermissionsGridViewImpl.java

```diff
@@ -99,37 +99,44 @@ private TableRow createAclEntryTableRow(final AclEntry aclEntry, PermissionLevel
 		permListBox.addStyleName("input-xs");
 		data = new TableData();
 		row.add(data);
-		if (!deleteButtonVisible || deleteButtonCallback == null) {
-			// Don't allow editing the permissions and don't add delete button.
-			String selectedItemText = permListBox.getSelectedItemText();
-			// SWC-6083: If this is the ACL entry of the public group
-			// and this is not editable (deleteButtonCallback == null),
-			// and the ACL entry is set to Can View,
-			// and this entity is set to be OPEN_DATA
-			// then render Can Download instead.
-			String canViewDisplayText = permissionDisplay.get(PermissionLevel.CAN_VIEW);
-			if (aclEntry.getOwnerId().equals(publicAclPrincipalId) && selectedItemText.equals(canViewDisplayText) && isOpenData) {
-				data.add(new Text(permissionDisplay.get(PermissionLevel.CAN_DOWNLOAD)));
+		boolean isDeleteButtonAvailable = deleteButtonVisible && deleteButtonCallback != null;
+		Widget permissionWidget;
+		
+		if (aclEntry.getOwnerId().equals(publicAclPrincipalId)) {
+			// SWC-6314 (and SWC-6083): If this is the public group, render Can Download if OPEN_DATA, otherwise Can View.
+			if (isOpenData) {
+				permissionWidget = new Text(permissionDisplay.get(PermissionLevel.CAN_DOWNLOAD));
 			} else {
-				data.add(new Text(selectedItemText));	
+				permissionWidget = new Text(permListBox.getSelectedItemText());
```

Comment:
Maybe missing something... Should you not be able to edit if you're admin?

Reply:
Usually yes.  However, this is the "Anyone on the web" entry in the ACL, and the backend only allows the READ permission (rendered as "Can view").

Reply:
Got it, thought it'd be needed to fix a bad permission given to public...